src/app/containers/NewSignupContainer/AccountStep/AccountForm.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Input, PasswordInput, PrimaryButton, Field, useApi, Row, Label, EmailInput, Block } from 'react-components';
+import { c } from 'ttag';
+import { queryCheckUsernameAvailability } from 'proton-shared/lib/api/user';
+
+const AccountForm = ({ onSubmit }) => {
+    const api = useApi();
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [passwordConfirmation, setPasswordConfirmation] = useState('');
+    const [email, setEmail] = useState('');
+    const [usernameError, setUsernameError] = useState();
+
+    const handleChangeUsername = ({ target }) => {
+        if (usernameError) {
+            setUsernameError(null);
+        }
+        setUsername(target.value);
+    };
+
+    const handleChangePassword = ({ target }) => setPassword(target.value);
+    const handleChangePasswordConfirmation = ({ target }) => setPasswordConfirmation(target.value);
+    const handleChangeEmail = ({ target }) => setEmail(target.value);
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        try {
+            await api(queryCheckUsernameAvailability(username));
+            onSubmit({
+                username,
+                password,
+                email
+            });
+        } catch (e) {
+            setUsernameError(e.data.Error);
+        }
+    };
+
+    return (
+        <form onSubmit={handleSubmit}>
```

Comment:
Use `useLoading` hook to disable submit form button please.

```suggestion
        <form onSubmit={() => withLoading(handleSubmit())}>
```